cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceQueryAdmin.java

```diff
@@ -103,11 +102,6 @@ public NamespaceMeta get(Id.Namespace namespaceId) throws Exception {
    */
   @Override
   public boolean exists(Id.Namespace namespaceId) throws Exception {
-    try {
-      get(namespaceId);
-    } catch (NotFoundException e) {
-      return false;
-    }
-    return true;
+    return nsStore.get(namespaceId) != null;
```

Comment:
hmmm, I'll see how to fix that. But I thought you hit the problem because ``listArtifacts(NamespaceId namespace)`` required access on ``namespace``, which this will fix.